### PR TITLE
Fixed bound errors, upgraded flutter map to 3.1.0 and fixed gpx exporting problem

### LINKED ISSUE #5: GPX exporting will export no coordinates when the bounds are not set
DatabaseDao.getCoordinatesWithBounds(int recordingId) will return no coordinates when the bounds are not set. It actually should return all coordinates.

### LINKED ISSUE #6: Upgrade to flutter_map 3.1.0 and fix zoom
When using flutter_map 3.0.0 I could just update the bounds of the map by updating the widget state. This is not possible in flutter_map 3.1.0. I've talked to one of the maintainers of flutter_map on Discord who was not sure if this was even expected behavior in the first place. But since 3.0.0 had other issues, we have to upgrade regardless. The maintainer told me that I should use a MapController and the fitBounds function instead, which is annoying. Will probably botch this.

For Reference: https://github.com/fleaflet/flutter_map/issues/1362.

### PR DESCRIPTION
**Pull request by TKFRvisionOfficial**
Closes #5, closes #6 

The bounds update after the device orientation changes only after the slider is moved again.

### CODE REVIEW

**Comment by TKFRvisionOfficial** [comment]
@revol-xut can you give me your thoughts on the behavior of the bound control buttons before merging? I thought about disabling the touch control when enabling the auto-bound lock.

File: lib/pages/recording_editor.dart [hunk 1]
```diff
@@ -185,47 +185,121 @@ class _RecordingEditorMap extends StatelessWidget {
...
     (Unsupported operation: Infinity or NaN toInt)
     This normally only happens during debugging.
-     */
+  */
 
-    if (bounds.southWest!.latitude == bounds.northEast!.latitude) {
-      bounds.southWest!.latitude += 0.0005;
-      bounds.northEast!.latitude -= 0.0005;
-    }
+  if (bounds.southWest!.latitude == bounds.northEast!.latitude) {
```
Comment (revol-xut) [inline lib/pages/recording_editor.dart:201]: I dont really understand this check to be honest. Does this mean that the window has zero width or height ?

**Review by revol-xut — APPROVED** [review]
did not see any flaws

File: lib/pages/recording_editor.dart (hunk 1 above)
Comment (TKFRvisionOfficial) [reply lib/pages/recording_editor.dart:201]: Well the bounds have 0 distance between them. And I can't tell you exactly but either flutter_map thinks the coordinates wrap around the whole world or it is actually a zero width issue.

**Pull request merged**